dali/operators/util/randomizer.cuh

```diff
@@ -64,6 +64,70 @@ struct curand_normal_dist<double> {
   }
 };
 
+
+template <typename T>
+struct curand_uniform_dist {};
+
+template <>
+struct curand_uniform_dist<float> {
+  DALI_HOST_DEV curand_uniform_dist(float start, float end)
+    : range_start_(start), range_size_(end-start) {}
+
+  DALI_HOST_DEV curand_uniform_dist()
+    : range_start_(0.0f), range_size_(1.0f) {}
+
+  __device__ inline float operator()(curandState *state) const {
+    return range_start_ + curand_uniform(state) * range_size_;
+  }
+ private:
+  float range_start_, range_size_;
+};
+
+template <>
+struct curand_uniform_dist<double> {
+  DALI_HOST_DEV curand_uniform_dist(float start, float end)
+    : range_start_(start), range_size_(end-start) {}
+
+  DALI_HOST_DEV curand_uniform_dist()
+    : range_start_(0.0f), range_size_(1.0f) {}
+
+  __device__ inline double operator()(curandState *state) const {
+    return range_start_ + curand_uniform_double(state) * range_size_;
+  }
+
+ private:
+  double range_start_, range_size_;
+};
+
+struct curand_uniform_int_range_dist {
+  DALI_HOST_DEV curand_uniform_int_range_dist(int start, int end)
+    : range_start_(start), range_size_(end-start) {}
+
+  __device__ inline int operator()(curandState *state) const {
+    return range_start_ + (curand(state) % range_size_);
+  }
+
+ private:
+  int range_start_;
+  unsigned int range_size_;
+};
+
+template <typename T>
+struct curand_uniform_int_values_dist {
+ public:
+  DALI_HOST_DEV curand_uniform_int_values_dist() : values_(nullptr), nvalues_(0) {}
```

Comment:
Would this be a segfault?
```c++
curand_uniform_int_values_dist<int> d;
curandState s;
d(&s);
```

Reply:
It would be division by zero, because `nvalues_(0)`. 
Yeah, this default constructor probably doesn't have any use, so I guess it can be removed.

Reply:
That's true but it is never used this way. The base implementation is written in generic terms and some of the distributions allow for a "default distribution" (default constructed). It is never used for this case though, but if I don't provide the default ctor here the code would become much messier.

What I can do is add an assert and a note that it should never be used

Reply:
If we rely on the fact that default constructed dist is never called then maybe an `assert(nvalues_)` can be added to the `operator()`. Doesn't change much because it will still crash, but al least in a way that shows that we predicted it.